### PR TITLE
TST: run e2e tests on ubuntu-latest

### PR DESCRIPTION
**Pull request by limpbrains**

### CODE REVIEW

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/w4lw47kk4372m64fmxk2satprm?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/1351)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/dzobxn54inojbpq4hs6q27myk4?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/1352)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/2j7klct3i5momgra7h34bywiji?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/1356)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/cvnvozp7ettvf6363imzrqmxti?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/1357)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/gsbf2544u3p5dummsfsmrseai4?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/1360)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/eqsadqlraj3glrzmvhrw5kg7wq?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/1361)

**Comment by Overtorment** [comment]
```

Invalid workflow file: .github/workflows/ci.yml#L62
You have an error in your yaml syntax on line 62

```

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/qklosxl5il2ghs4bk4puohricq?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/1400)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/j4bnaisxq7seumpihrh24y24my?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/1401)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/t7pcxeva54oskfe6rih5l2iege?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/1402)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/mr66w44tmpondbpg5avx2yoyv4?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/1403)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/xj63vvgj2fwaqut5t3il5idigm?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/1404)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/agzsajkexgldvncmmvh2q3eicm?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/1405)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/bheltkqirqwbece67ibawy6w6u?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/1409)

File: tests/e2e/bluewallet2.spec.js [hunk 1]
```diff
@@ -46,7 +46,7 @@ describe('BlueWallet UI Tests - import BIP84 wallet', () => {
 
     // setting fee rate:
     const feeRate = 2;
-    await element(by.id('chooseFee')).tap();
+    await element(by.id('chooseFee1')).tap();
```
Comment (Overtorment) [inline tests/e2e/bluewallet2.spec.js:49]: is that on purpose to make it fail?

Comment (limpbrains) [reply tests/e2e/bluewallet2.spec.js:49]: Yeah, testing artifact uploading

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/gs6j4gsap444a4744kt2homy7y?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/1423)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/ed4nf5uflv7ie6fujubng5krvm?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/1426)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/bcefrws7dpatwl5t5pduoxznhe?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/1428)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/7rc3py7fcep7t32yd7ur4girli?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/1431)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/l2kk4ke3664fkrfrlw5kaluruq?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/1432)

**Comment by limpbrains** [comment]
runs stable
~ 40 min average

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -84,68 +115,35 @@ jobs:
           ${{ runner.os }}-npm-
 
     - name: Install node_modules
-      run: npm install
+      run: npm install || npm install
```
Comment (Overtorment) [inline .github/workflows/ci.yml:118]: wut

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -55,10 +55,41 @@ jobs:
...
+    - name: npm and gradle caches in /mnt
+      run: |
+        rm -rf ~/.npm
+        rm -rf ~/.gradle
+        sudo mkdir -p /mnt/.npm
+        sudo mkdir -p /mnt/.gradle
+        sudo chown -R runner /mnt/.npm
+        sudo chown -R runner /mnt/.gradle
+        ln -s /mnt/.npm /home/runner/
+        ln -s /mnt/.gradle /home/runner/
```
Comment (Overtorment) [inline .github/workflows/ci.yml:87]: looks a bit fragile tbh

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -84,68 +115,35 @@ jobs:
...
-      with:
-        api-level: 31
-        avd-name: Pixel_API_29_AOSP
-        force-avd-creation: false
-        emulator-options: -no-window -gpu swiftshader_indirect -no-snapshot -noaudio -no-boot-anim -camera-back none -camera-front none -partition-size 2047
-        arch: x86_64
-        script: npm run e2e:release-test
-
-    - name: Test attempt 4
+    - name: Run tests
```
Comment (Overtorment) [inline .github/workflows/ci.yml:135]: so you threw away all test attempts you coded? in favor of `-R 3` key?
is the output redable? you like it?

File: package.json [hunk 5]
```diff
@@ -64,7 +64,7 @@
     "e2e:debug-test": "detox test -c android.debug -d 200000 -l info",
     "e2e:debug": "(test -f android/app/build/outputs/apk/debug/app-debug.apk && test -f android/app/build/outputs/apk/androidTest/debug/app-debug-androidTest.apk) || npm run e2e:debug-build; npm run e2e:debug-test",
     "e2e:release-build": "detox build -c android.release",
-    "e2e:release-test": "detox test -c android.release --record-videos all --record-logs all --take-screenshots all --headless -d 200000 -R 3",
+    "e2e:release-test": "detox test -c android.release",
```
Comment (Overtorment) [inline package.json:67]: i suppose you want these arguments to be CI-specific, while on localhost this npm command just runs plain single e2e  run..?

File: .github/workflows/ci.yml [hunk 6]
```diff
@@ -55,10 +55,41 @@ jobs:
...
+        rm -rf ~/.npm
+        rm -rf ~/.gradle
+        sudo mkdir -p /mnt/.npm
+        sudo mkdir -p /mnt/.gradle
+        sudo chown -R runner /mnt/.npm
+        sudo chown -R runner /mnt/.gradle
+        ln -s /mnt/.npm /home/runner/
+        ln -s /mnt/.gradle /home/runner/
+
+    - name: Create artifacts directory on /mnt
```
Comment (Overtorment) [inline .github/workflows/ci.yml:89]: btw do we know whats this mount point..?

Comment (Overtorment) [reply .github/workflows/ci.yml:89]: where does it point exactly?

**Review by Overtorment — CHANGES_REQUESTED** [review]
great job overal! just have some questions.
plus im kinda interested to compare old e2e on macos to this ubuntu-based back-to-back

**Comment by Overtorment** [comment]
man i went through a bunch or recent PRs, and their e2es all take at least 80 min. sometimes 120, 200 min.
the difference is YUUUUUUUUUUUUGE

**Comment by Overtorment** [comment]
why theres such a difference??

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/niyobnv4wfnnfgpsmvsi277w3e?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/1439)

File: .github/workflows/ci.yml (hunk 2 above)
Comment (limpbrains) [reply .github/workflows/ci.yml:118]: because npm fails sometimes, like error 400 or something

File: .github/workflows/ci.yml (hunk 3 above)
Comment (limpbrains) [reply .github/workflows/ci.yml:87]: look ok to me

File: .github/workflows/ci.yml (hunk 4 above)
Comment (limpbrains) [reply .github/workflows/ci.yml:135]: -R 3 is simplier, that is why I'm choosing it

File: package.json (hunk 5 above)
Comment (limpbrains) [reply package.json:67]: exectly. You don't ususally record a video on localhost, or run test 3 times

File: .github/workflows/ci.yml (hunk 6 above)
Comment (limpbrains) [reply .github/workflows/ci.yml:89]: No idea. Found that people who have similar issues use it.
Some temporary storage, I assume

**Comment by limpbrains** [comment]
> why theres such a difference??

Who knows 🤷
I think they just finally fixed virtualisation for Linux runners. For macos VM tools always have been poor 

Will see how arm runners will work

**Review by Overtorment — APPROVED** [review]

**Comment by GladosBlueWallet** [comment]
Unbelievable. You, [subject name here], must be the pride of [subject hometown here]!

**Pull request merged**